pkg/metricsadapter/provider/resourcemetrics.go

```diff
@@ -0,0 +1,636 @@
+type NodeLister struct {
+	clusterLister   clusterlister.ClusterLister
+	informerManager genericmanager.MultiClusterInformerManager
+}
+
+// NewNodeLister creates an internal new NodeLister
+func NewNodeLister(clusterLister clusterlister.ClusterLister, informerManager genericmanager.MultiClusterInformerManager) *NodeLister {
+	return &NodeLister{
+		clusterLister:   clusterLister,
+		informerManager: informerManager,
+	}
+}
+
+// List returns the internal constructed node with label selector info
+func (n *NodeLister) List(selector labels.Selector) (ret []*corev1.Node, err error) {
+	klog.V(4).Infof("Query node metrics with selector: %s", selector.String())
+
+	clusters, err := n.clusterLister.List(labels.Everything())
+	if err != nil {
+		return nil, err
+	}
+
+	for _, cluster := range clusters {
+		sci := n.informerManager.GetSingleClusterManager(cluster.Name)
+		if sci == nil {
+			klog.Errorf("Failed to get SingleClusterInformerManager for cluster(%s)", cluster.Name)
+			continue
+		}
+		nodes, err := sci.Lister(NodesGVR).List(selector)
+		if err != nil {
+			klog.Infof("Failed to list nodes from cluster(%s): %v", cluster.Name, err)
+			return nil, err
+		}
+		for index := range nodes {
+			nodeTyped := &corev1.Node{}
+			err = helper.ConvertToTypedObject(nodes[index], nodeTyped)
+			if err != nil {
+				klog.Errorf("Failed to convert to typed object: %v", err)
+				return nil, err
+			}
+			if nodeTyped.Annotations == nil {
+				nodeTyped.Annotations = map[string]string{}
+			}
+
+			//If user sets this annotation, we need to remove it to avoid parsing wrong next.
+			nodeTyped.Annotations[labelSelectorAnnotationInternal] = selector.String()
+			ret = append(ret, nodeTyped)
+		}
+	}
+
+	return ret, nil
+}
+
+// Get returns the internal constructed node with name info
+func (n *NodeLister) Get(name string) (*corev1.Node, error) {
+	klog.V(4).Infof("Query node metrics with name:%s", name)
+
+	clusters, err := n.clusterLister.List(labels.Everything())
+	if err != nil {
+		return nil, err
+	}
+
+	for _, cluster := range clusters {
+		sci := n.informerManager.GetSingleClusterManager(cluster.Name)
+		if sci == nil {
+			klog.Errorf("Failed to get SingleClusterInformerManager for cluster(%s)", cluster.Name)
+			continue
+		}
+		node, err := sci.Lister(NodesGVR).Get(name)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				klog.Infof("Failed to get node from cluster(%s):%v", cluster.Name, err)
+			}
+			continue
+		}
+		// We only return the first matched cluster
+		nodeTyped := &corev1.Node{}
+		err = helper.ConvertToTypedObject(node, nodeTyped)
+		if err != nil {
+			klog.Errorf("Failed to convert to typed object: %v", err)
+			return nil, err
+		}
+		if nodeTyped.Annotations == nil {
+			nodeTyped.Annotations = map[string]string{}
+		}
+
+		//If user sets this annotation, we need to remove it to avoid parsing wrong next.
+		delete(nodeTyped.Annotations, labelSelectorAnnotationInternal)
+		return nodeTyped, nil
```

Comment:
Will there be nodes with the same name in multiple member clusters?

Reply:
It's possible, But if query node metrics with name, I think we only can return the first matched clusters. What do you think?

Reply:
Talked with @chaunceyjiang , If there will be multiple resources with the same name for the get method. An error will be returned.